src/Microsoft.AspNetCore.Razor.Evolution/DefaultRazorRuntimeCSharpLoweringPhase.cs

```diff
@@ -332,6 +332,16 @@ internal override void VisitCreateTagHelper(CreateTagHelperIRNode node)
                     tagHelperVariableName);
             }
 
+            internal override void VisitAddPreallocatedTagHelperHtmlAttribute(AddPreallocatedTagHelperHtmlAttributeIRNode node)
```

Comment:
Agree. We can move nearly all of the TagHelper IR nodes into `CSharpStatementIRNode`s which would make these pre-allocated IR bits unneeded.